Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -0,0 +1,2978 @@
+
+MarvinDataSgroup::MarvinDataSgroup(MarvinMolBase *parentInit) {
+  parent = parentInit;
+}
+
+MarvinMolBase *MarvinDataSgroup::copyMol(std::string idAppendage) const {
+  auto outSgroup = new MarvinDataSgroup(this->parent);
+  this->parent->sgroups.push_back(outSgroup);
+
+  outSgroup->molID = this->molID + idAppendage;
+  outSgroup->id = this->id + idAppendage;
+  outSgroup->context = this->context;
+  outSgroup->fieldName = this->fieldName;
+  outSgroup->placement = this->placement;
+  outSgroup->unitsDisplayed = this->unitsDisplayed;
+  outSgroup->units = this->units;
+  outSgroup->fieldData = this->fieldData;
+  outSgroup->queryType = this->queryType;
+  outSgroup->queryOp = this->queryOp;
+  outSgroup->x = this->x;
+  outSgroup->y = this->y;
+
+  // the only time this is to be called is when a mutliple group above it is
+  // being expanded and the new atoms and bonds have already been made
+  //  here we just need to find them and add refs to the new multiple sgroup
+
+  for (auto atomToCopy : atoms) {
+    auto atomIndex = this->parent->getAtomIndex(atomToCopy->id + idAppendage);
+    if (atomIndex < 0) {
+      throw FileParseException(
+          "unexpected error - new atom not found in parent");
+    }
+
+    outSgroup->atoms.push_back(this->parent->atoms[atomIndex]);
+  }
+
+  return outSgroup;
+}
+
+std::string MarvinDataSgroup::toString() const {
+  std::ostringstream out;
+
+  out << "<molecule molID=\"" << molID << "\" id=\"" << id
+      << "\" role=\"DataSgroup\" atomRefs=\""
+      << boost::algorithm::join(getAtomList(), " ") << "\" context=\""
+      << context << "\" fieldName=\"" << fieldName << "\" placement=\""
+      << placement << "\" unitsDisplayed=\"" << unitsDisplayed
+      << "\" fieldData=\"" << fieldData;
+
+  if (units != "") {
+    out << "\" units=\"" << units;
+  }
+
+  if (queryType != "" && queryOp != "") {
+    out << "\" queryType=\"" << queryType << "\" queryOp=\""
+        << boost::property_tree::xml_parser::encode_char_entities(queryOp);
+  }
+
+  out << "\" x=\"" << x << "\" y=\"" << y << "\"/>";
+
+  return out.str();
+}
+
+ptree MarvinDataSgroup::toPtree() const {
+  ptree out = MarvinMolBase::toPtree();
+
+  out.put("<xmlattr>.atomRefs", boost::algorithm::join(getAtomList(), " "));
+  out.put("<xmlattr>.context", context);
+  out.put("<xmlattr>.fieldName", fieldName);
+  out.put("<xmlattr>.placement", placement);
+  out.put("<xmlattr>.unitsDisplayed", unitsDisplayed);
+  out.put("<xmlattr>.fieldData", fieldData);
+
+  if (units != "") {
+    out.put("<xmlattr>.units", units);
+  }
+
+  if (queryType != "" && queryOp != "") {
+    out.put("<xmlattr>.queryType", queryType);
+    out.put("<xmlattr>.queryOp", queryOp);
+  }
+  std::ostringstream xstr, ystr;
+  xstr << x;
+  ystr << y;
+  out.put("<xmlattr>.x", xstr.str());
+  out.put("<xmlattr>.y", ystr.str());
+
+  return out;
+}
+
+std::string MarvinDataSgroup::role() const { return std::string("DataSgroup"); }
```

Comment:
I remove the string constructors.  But since these rest are all literals (e.g. "MultipleSgroup"), the constructor will still be called (unless the compiler is smart enough to do it at compile time)